### PR TITLE
WT-10235 Update code, tests and doc related to packing and unpacking data

### PR DESCRIPTION
**Pull request by etienneptl**

### CODE REVIEW

File: src/docs/packing.dox [hunk 1]
```diff
@@ -20,8 +20,7 @@ If the first character is not one of these, '.' (big-endian, packed) is
 assumed: it naturally sorts in lexicographic order, and the packed format
 uses variable-sized encoding of values to reduce the data size.
 
-@notyet{little-endian format}
-Only the default big-endian, packed format is currently supported.
+Note: Only the default big-endian, packed format is currently supported.
```
Comment (etienneptl) [inline src/docs/packing.dox:23]: I compiled the doc and checked this sentence correctly appears.

**Review by sulabhM — APPROVED** [review]
LGTM

File: test/packing/packing-test.c [hunk 2]
```diff
@@ -32,48 +32,61 @@
  * check --
  *     TODO: Add a comment describing this function.
```
[the remark this answers is not in the document]
Comment (etienneptl) [reply test/packing/packing-test.c]: Done in aaf5eae9d537a58825dbf640e01157054a268e6c

**Comment by etienneptl** [comment]
The error in Evergreen is not related to those changes.

**Pull request merged**